### DISCUSSION TITLE
Add phonemizer for custom dataset

### DISCUSSION THREAD

**Discussion by huydang2106** [coqui-ai/TTS#2181] [General Q&A]
I want to train VITS model for my own dataset. I have built my phonemizer and intergrate it in training process. But it produces quite bad performance (although it is quite good when I used it for training another model with Espnet), so I think it maybe have some problems when intergrating the phonemizer. Is there any documentation about building custom phonemizer with coqui ai tts ?

**Comment by thorstenMueller** [comment]
Why building your own phonemizer? Isn't your language supported by eSpeak?

**Reply by huydang2106** [reply]
eSpeak can support many language, but there are still some dialects of those language which are not  covered. Also I think my custom phonemizer can produce better utterances than the default of eSpeak

**Reply by huydang2106** [reply]
Also, when I use eSpeak for my dataset (vietnamese), it constantly produces OOV tokens. So it comes to my mind that better to build a custom phonemizer for each language

**Comment by Aml-Hassan-Abd-El-hamid** [comment]
any update on that, I'm trying to work on the Egyptian dialect.
